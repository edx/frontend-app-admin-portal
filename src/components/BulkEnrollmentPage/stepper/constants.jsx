export const STEPPER_TITLE = 'Learner enrollment';
export const ADD_LEARNERS_TITLE = 'Add learners';
export const REVIEW_TITLE = 'Review';
export const NEXT_BUTTON_TEXT = 'Next';
export const PREVIOUS_BUTTON_TEXT = 'Previous';
export const FINAL_BUTTON_TEXT = 'Enroll learners';
export const NEXT_BUTTON_TEST_ID = 'stepper-next';
export const PREV_BUTTON_TEST_ID = 'stepper-prev';
export const FINAL_BUTTON_TEST_ID = 'stepper-final';
