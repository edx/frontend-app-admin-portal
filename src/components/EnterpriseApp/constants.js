/* eslint-disable import/prefer-default-export */

export const ROUTE_NAMES = {
  bulkEnrollment: 'catalog-management',
  subscriptionManagement: 'subscriptions',
};
