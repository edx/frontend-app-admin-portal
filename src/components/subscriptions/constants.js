export const TAB_ALL_USERS = 'TAB_ALL_USERS';
export const TAB_LICENSED_USERS = 'TAB_LICENSED_USERS';
export const TAB_PENDING_USERS = 'TAB_PENDING_USERS';
export const TAB_REVOKED_USERS = 'TAB_REVOKED_USERS';

export const PAGE_SIZE = 10;

// Subscription license statuses as defined on the backend
export const ACTIVATED = 'activated';
export const ASSIGNED = 'assigned';
export const REVOKED = 'revoked';
export const ALL_USERS = 'assigned,activated,revoked';

export const SUBSCRIPTIONS = 'Subscriptions';
export const SUBSCRIPTION_OVERVIEW = 'Subscription Overview';
export const SUBSCRIPTION_USERS = 'Subscription Users';
export const SUBSCRIPTION_USERS_OVERVIEW = 'Subscription Users Overview';

export const NETWORK_ERROR_MESSAGE = 'Error occurred while loading the data.';
export const DEFAULT_PAGE = 1;

// used to determine whether to show the revocation cap messaging
export const SHOW_REVOCATION_CAP_PERCENT = 10;
