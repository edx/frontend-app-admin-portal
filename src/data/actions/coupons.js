import {
  COUPONS_REQUEST,
  COUPONS_SUCCESS,
  COUPONS_FAILURE,
  CLEAR_COUPONS,
} from '../constants/coupons';

// import { updateUrl } from '../../utils';

import EcommerceApiService from '../services/EcommerceApiService';

const fetchCouponOrdersRequest = () => ({
  type: COUPONS_REQUEST,
});

const fetchCouponOrdersSuccess = data => ({
  type: COUPONS_SUCCESS,
  payload: {
    data,
  },
});

const fetchCouponOrdersFailure = error => ({
  type: COUPONS_FAILURE,
  payload: {
    error,
  },
});

const clearCouponOrdersEvent = () => ({ type: CLEAR_COUPONS });

const fetchCouponOrders = options => (
  (dispatch) => {
    dispatch(fetchCouponOrdersRequest(options));
    return EcommerceApiService.fetchCouponOrders(options)
      .then((response) => {
        dispatch(fetchCouponOrdersSuccess(response.data));
      })
      .catch((error) => {
        // This endpoint returns a 404 if no data exists,
        // so we convert it to an empty response here.
        if (error.response.status === 404) {
          dispatch(fetchCouponOrdersSuccess({ results: [] }));
          return;
        }
        dispatch(fetchCouponOrdersFailure(error));
      });
  }
);

const clearCouponOrders = () => (
  (dispatch) => {
    dispatch(clearCouponOrdersEvent());
  }
);

export {
  fetchCouponOrders,
  clearCouponOrders,
};
