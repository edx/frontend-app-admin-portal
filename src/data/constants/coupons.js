const COUPONS_REQUEST = 'COUPONS_REQUEST';
const COUPONS_SUCCESS = 'COUPONS_SUCCESS';
const COUPONS_FAILURE = 'COUPONS_FAILURE';
const CLEAR_COUPONS = 'CLEAR_COUPONS';

export {
  COUPONS_REQUEST,
  COUPONS_SUCCESS,
  COUPONS_FAILURE,
  CLEAR_COUPONS,
};
