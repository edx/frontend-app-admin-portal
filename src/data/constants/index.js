// eslint-disable-next-line import/prefer-default-export
export const ENTERPRISE_ADMIN_ROLE_NAME = 'enterprise_admin';
