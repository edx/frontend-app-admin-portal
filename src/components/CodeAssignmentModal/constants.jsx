import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
import { MODAL_TYPES } from '../EmailTemplateForm/constants';
import { EMAIL_TEMPLATE_FIELDS } from '../EmailTemplateForm';
import CheckboxWithTooltip from '../ReduxFormCheckbox/CheckboxWithTooltip';

export const ASSIGNMENT_ERROR_TITLES = {
  [MODAL_TYPES.assign]: 'Unable to assign codes',
  [MODAL_TYPES.save]: 'Unable to save template',
};
export const EMAIL_TEMPLATE_NUDGE_EMAIL_ID = 'enable-nudge-emails';

export const ASSIGNMENT_MODAL_FIELDS = {
  ...EMAIL_TEMPLATE_FIELDS,
  [EMAIL_TEMPLATE_NUDGE_EMAIL_ID]: {
    name: EMAIL_TEMPLATE_NUDGE_EMAIL_ID,
    id: EMAIL_TEMPLATE_NUDGE_EMAIL_ID,
    component: CheckboxWithTooltip,
    className: 'auto-reminder-wrapper',
    icon: faInfoCircle,
    altText: 'More information',
    tooltipText: 'edX will remind learners to redeem their code 3, 10, and 19 days after you assign it.',
    label: 'Automate reminders',
  },
};
