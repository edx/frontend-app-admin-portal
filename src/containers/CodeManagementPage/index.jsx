import { connect } from 'react-redux';

import CodeManagement from '../../components/CodeManagement';

import { fetchCouponOrders, clearCouponOrders } from '../../data/actions/coupons';
import { fetchPortalConfiguration } from '../../data/actions/portalConfiguration';

const mapStateToProps = state => ({
  loading: state.coupons.loading,
  error: state.coupons.error,
  enterpriseId: state.portalConfiguration.enterpriseId,
  coupons: state.coupons.data,
});

const mapDispatchToProps = dispatch => ({
  fetchPortalConfiguration: (slug) => {
    dispatch(fetchPortalConfiguration(slug));
  },
  fetchCouponOrders: () => {
    dispatch(fetchCouponOrders());
  },
  clearCouponOrders: () => {
    dispatch(clearCouponOrders());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(CodeManagement);
