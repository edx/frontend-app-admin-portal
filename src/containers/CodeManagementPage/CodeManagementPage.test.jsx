import React from 'react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';

import CodeManagementPage from './index';

import { COUPONS_REQUEST, CLEAR_COUPONS } from '../../data/constants/coupons';

const mockStore = configureMockStore([thunk]);
const initialState = {
  portalConfiguration: {
    enterpriseId: 'test-enterprise-id',
  },
  coupons: {
    loading: false,
    error: null,
    data: {
      count: 0,
      results: [],
    },
  },
  table: {
    'coupon-details': {},
  },
};

const CodeManagementPageWrapper = props => (
  <MemoryRouter>
    <Provider store={props.store}>
      <CodeManagementPage
        location={{}}
        match={{
          path: '/test-page',
        }}
        history={{
          replace: () => {},
        }}
        {...props}
      />
    </Provider>
  </MemoryRouter>
);

CodeManagementPageWrapper.defaultProps = {
  store: mockStore({ ...initialState }),
};

CodeManagementPageWrapper.propTypes = {
  store: PropTypes.shape({}),
};

const sampleCouponData = {
  id: 0,
  title: 'test-title-1',
  start_date: '2019-01-03T23:23:51.581Z',
  end_date: '2019-09-18T14:46:36.716Z',
  has_error: false,
  max_uses: 10,
  num_unassigned: 2,
  num_uses: 2,
};

describe('CodeManagementPageWrapper', () => {
  describe('renders', () => {
    it('renders empty results correctly', () => {
      const tree = renderer
        .create((
          <CodeManagementPageWrapper />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders non-empty results correctly', () => {
      const store = mockStore({
        ...initialState,
        coupons: {
          ...initialState.coupons,
          data: {
            count: 2,
            num_pages: 1,
            results: [
              sampleCouponData,
              {
                ...sampleCouponData,
                id: 1,
                title: 'test-title-2',
              },
            ],
          },
        },
      });

      const tree = renderer
        .create((
          <CodeManagementPageWrapper store={store} />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders loading state correctly', () => {
      const store = mockStore({
        ...initialState,
        coupons: {
          ...initialState.coupons,
          loading: true,
        },
      });

      const tree = renderer
        .create((
          <CodeManagementPageWrapper store={store} />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders error state correctly', () => {
      const store = mockStore({
        ...initialState,
        coupons: {
          ...initialState.coupons,
          error: new Error('test error'),
        },
      });

      const tree = renderer
        .create((
          <CodeManagementPageWrapper store={store} />
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('correctly handles prop changes', () => {
    it('handles location.state change', () => {
      const wrapper = mount(<CodeManagementPageWrapper />);

      wrapper.setProps({
        location: {
          state: {
            hasRequestedCodes: true,
          },
        },
      });

      expect(wrapper.find('CodeManagement').instance().state.hasRequestedCodes).toBeTruthy();
    });
  });

  it('calls clearCouponOrders() on componentWillUnmount', () => {
    const store = mockStore({
      ...initialState,
    });

    const wrapper = mount(<CodeManagementPageWrapper store={store} />);
    wrapper.unmount();

    const actions = store.getActions();
    expect(actions.filter(action => action.type === CLEAR_COUPONS)).toHaveLength(1);
  });

  it('calls expand/collapse callbacks properly', () => {
    const store = mockStore({
      ...initialState,
      coupons: {
        ...initialState.coupons,
        data: {
          count: 1,
          num_pages: 1,
          results: [
            sampleCouponData,
            {
              ...sampleCouponData,
              id: 1,
              title: 'test-title-2',
            },
          ],
        },
      },
      table: {
        'coupon-details': {},
      },
      csv: {
        'coupon-details': {},
      },
    });

    const wrapper = mount(<CodeManagementPageWrapper store={store} />);
    const instance = wrapper.find('CodeManagement').instance();
    const spyExpand = jest.spyOn(instance, 'handleCouponExpand');
    const spyCollapse = jest.spyOn(instance, 'handleCouponCollapse');

    // expand
    wrapper.find('Coupon').first().find('.metadata').simulate('click');
    expect(spyExpand).toBeCalledTimes(1);

    // collapse
    wrapper.find('Coupon').first().find('.metadata').simulate('click');
    expect(spyCollapse).toBeCalledTimes(1);
  });

  it('fetches coupons on refresh button click', () => {
    const store = mockStore({ ...initialState });
    const wrapper = mount(<CodeManagementPageWrapper store={store} />);
    store.clearActions();
    wrapper.find('.fa-refresh').simulate('click');
    expect(store.getActions().filter(action => action.type === COUPONS_REQUEST)).toHaveLength(1);
  });
});
