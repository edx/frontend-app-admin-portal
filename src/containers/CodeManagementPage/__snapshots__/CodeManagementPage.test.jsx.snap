// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeManagementPageWrapper renders renders empty results correctly 1`] = `
Array [
  <div
    className="hero py-4 pt-md-5 pb-md-0"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-md-9 mb-2 mb-md-1"
        >
          <h1
            className={null}
          >
            Code Management
          </h1>
        </div>
        <div
          className="col-12 col-md-3 text-md-right mb-2 mb-md-1"
        >
          <img
            alt="edX for Business"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container-fluid"
  >
    <div
      className="row mt-4 mb-3"
    >
      <div
        className="col-12 col-md-3 mb-2 mb-md-0"
      >
        <h2
          className={null}
        >
          Overview
        </h2>
      </div>
      <div
        className="col-12 col-md-9 mb-2 mb-md-0 text-md-right"
      >
        <button
          className="btn mr-2 btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa fa-refresh mr-2"
            id="Icon1"
          />
          Refresh data
        </button>
        <a
          className="request-codes-btn btn btn-primary"
          href="/test-page/request"
          onClick={[Function]}
        >
          <span
            aria-hidden={true}
            className="fa fa-plus mr-2"
            id="Icon1"
          />
          Request more codes
        </a>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="alert fade alert-warning show"
          hidden={false}
          role="alert"
        >
          <div
            className="alert-dialog"
          >
            <div
              className="d-flex"
            >
              <div
                className="icon mr-2"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-exclamation-circle"
                  id="Icon1"
                />
              </div>
              <div
                className="message"
              >
                There are no results.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`CodeManagementPageWrapper renders renders error state correctly 1`] = `
Array [
  <div
    className="hero py-4 pt-md-5 pb-md-0"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-md-9 mb-2 mb-md-1"
        >
          <h1
            className={null}
          >
            Code Management
          </h1>
        </div>
        <div
          className="col-12 col-md-3 text-md-right mb-2 mb-md-1"
        >
          <img
            alt="edX for Business"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container-fluid"
  >
    <div
      className="row mt-4 mb-3"
    >
      <div
        className="col-12 col-md-3 mb-2 mb-md-0"
      >
        <h2
          className={null}
        >
          Overview
        </h2>
      </div>
      <div
        className="col-12 col-md-9 mb-2 mb-md-0 text-md-right"
      >
        <button
          className="btn mr-2 btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa fa-refresh mr-2"
            id="Icon1"
          />
          Refresh data
        </button>
        <a
          className="request-codes-btn btn btn-primary"
          href="/test-page/request"
          onClick={[Function]}
        >
          <span
            aria-hidden={true}
            className="fa fa-plus mr-2"
            id="Icon1"
          />
          Request more codes
        </a>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="mt-3 alert fade alert-danger show"
          hidden={false}
          role="alert"
        >
          <div
            className="alert-dialog"
          >
            <div
              className="d-flex"
            >
              <div
                className="icon mr-2"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-times-circle"
                  id="Icon1"
                />
              </div>
              <div
                className="message"
              >
                <span
                  className="title"
                >
                  Unable to load coupons
                </span>
                Try refreshing your screen (test error)
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`CodeManagementPageWrapper renders renders loading state correctly 1`] = `
Array [
  <div
    className="hero py-4 pt-md-5 pb-md-0"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-md-9 mb-2 mb-md-1"
        >
          <h1
            className={null}
          >
            Code Management
          </h1>
        </div>
        <div
          className="col-12 col-md-3 text-md-right mb-2 mb-md-1"
        >
          <img
            alt="edX for Business"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container-fluid"
  >
    <div
      className="row mt-4 mb-3"
    >
      <div
        className="col-12 col-md-3 mb-2 mb-md-0"
      >
        <h2
          className={null}
        >
          Overview
        </h2>
      </div>
      <div
        className="col-12 col-md-9 mb-2 mb-md-0 text-md-right"
      >
        <button
          className="btn mr-2 btn-link"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa fa-refresh mr-2"
            id="Icon1"
          />
          Refresh data
        </button>
        <a
          className="request-codes-btn btn btn-primary"
          href="/test-page/request"
          onClick={[Function]}
        >
          <span
            aria-hidden={true}
            className="fa fa-plus mr-2"
            id="Icon1"
          />
          Request more codes
        </a>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="loading d-flex align-items-center justify-content-center coupons mt-3"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`CodeManagementPageWrapper renders renders non-empty results correctly 1`] = `
Array [
  <div
    className="hero py-4 pt-md-5 pb-md-0"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12 col-md-9 mb-2 mb-md-1"
        >
          <h1
            className={null}
          >
            Code Management
          </h1>
        </div>
        <div
          className="col-12 col-md-3 text-md-right mb-2 mb-md-1"
        >
          <img
            alt="edX for Business"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="container-fluid"
  >
    <div
      className="row mt-4 mb-3"
    >
      <div
        className="col-12 col-md-3 mb-2 mb-md-0"
      >
        <h2
          className={null}
        >
          Overview
        </h2>
      </div>
      <div
        className="col-12 col-md-9 mb-2 mb-md-0 text-md-right"
      >
        <button
          className="btn mr-2 btn-link"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa fa-refresh mr-2"
            id="Icon1"
          />
          Refresh data
        </button>
        <a
          className="request-codes-btn btn btn-primary"
          href="/test-page/request"
          onClick={[Function]}
        >
          <span
            aria-hidden={true}
            className="fa fa-plus mr-2"
            id="Icon1"
          />
          Request more codes
        </a>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="coupon mb-3 mb-lg-2 rounded border"
        >
          <div
            aria-controls="coupon-details-0"
            aria-expanded={false}
            className="metadata row no-gutters p-2 d-flex align-items-center rounded"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
            >
              <small
                className="text-muted"
              >
                Coupon Name
              </small>
              <div>
                test-title-1
              </div>
            </div>
            <div
              className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col"
                >
                  <small
                    className="text-muted"
                  >
                    Valid From
                  </small>
                  <div>
                    January 3, 2019
                  </div>
                </div>
                <div
                  className="col"
                >
                  <small
                    className="text-muted"
                  >
                    Valid To
                  </small>
                  <div>
                    September 18, 2019
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col"
                >
                  <small
                    className="text-muted"
                  >
                    Unassigned Codes
                  </small>
                  <div>
                    2
                  </div>
                </div>
                <div
                  className="col"
                >
                  <small
                    className="text-muted"
                  >
                    Enrollments Redeemed
                  </small>
                  <div>
                    2 of 10
                    <span
                      className="ml-1"
                    >
                      (20%)
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
            >
              <span
                aria-hidden={true}
                className="fa fa-chevron-down text-primary"
                id="Icon1"
              />
              <span
                className="sr-only"
              >
                Open coupon details
              </span>
            </div>
          </div>
          <div
            className="coupon-details row no-gutters px-2 my-3 d-none"
            id="coupon-details-0"
          >
            <div
              className="col"
            />
          </div>
        </div>
        <div
          className="coupon mb-3 mb-lg-2 rounded border"
        >
          <div
            aria-controls="coupon-details-1"
            aria-expanded={false}
            className="metadata row no-gutters p-2 d-flex align-items-center rounded"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="col-sm-12 col-lg-3 mb-2 mb-lg-0"
            >
              <small
                className="text-muted"
              >
                Coupon Name
              </small>
              <div>
                test-title-2
              </div>
            </div>
            <div
              className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col"
                >
                  <small
                    className="text-muted"
                  >
                    Valid From
                  </small>
                  <div>
                    January 3, 2019
                  </div>
                </div>
                <div
                  className="col"
                >
                  <small
                    className="text-muted"
                  >
                    Valid To
                  </small>
                  <div>
                    September 18, 2019
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-sm-12 col-lg-4 mb-2 mb-lg-0"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col"
                >
                  <small
                    className="text-muted"
                  >
                    Unassigned Codes
                  </small>
                  <div>
                    2
                  </div>
                </div>
                <div
                  className="col"
                >
                  <small
                    className="text-muted"
                  >
                    Enrollments Redeemed
                  </small>
                  <div>
                    2 of 10
                    <span
                      className="ml-1"
                    >
                      (20%)
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="icons col-lg-1 order-first order-lg-last text-right pr-2 mt-1 m-lg-0"
            >
              <span
                aria-hidden={true}
                className="fa fa-chevron-down text-primary"
                id="Icon1"
              />
              <span
                className="sr-only"
              >
                Open coupon details
              </span>
            </div>
          </div>
          <div
            className="coupon-details row no-gutters px-2 my-3 d-none"
            id="coupon-details-1"
          >
            <div
              className="col"
            />
          </div>
        </div>
        <div
          className="d-flex mt-4 justify-content-center"
        >
          <nav
            aria-label="coupons pagination"
            className=""
          >
            <div
              aria-atomic={true}
              aria-live="polite"
              aria-relevant="text"
              className="sr-only"
            >
              Page 1, Current Page, of 1
            </div>
            <ul
              className="pagination"
            >
              <li
                className="page-item disabled"
              >
                <button
                  aria-label="Previous"
                  className="btn previous page-link"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  tabIndex="-1"
                  type="button"
                >
                  <div>
                    <span
                      aria-hidden={true}
                      className="fa fa-chevron-left mr-2"
                      id="pagination-3"
                    />
                    Previous
                  </div>
                </button>
              </li>
              <li
                className="page-item disabled"
              >
                <button
                  aria-label="Next"
                  className="btn next page-link"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  tabIndex="-1"
                  type="button"
                >
                  <div>
                    Next
                    <span
                      aria-hidden={true}
                      className="fa fa-chevron-right ml-2"
                      id="pagination-4"
                    />
                  </div>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>,
]
`;
